packages/api-reference/src/components/ApiReferenceLayout.vue

```diff
@@ -37,6 +38,17 @@ defineEmits<{
   (e: 'toggleDarkMode'): void
 }>()
 
+// Put the configuration in a globally accessible store
+const { setConfiguration } = useConfigurationStore()
```

Comment:
@hanspagel do we need to put the config in a global store?

I think we want to prop drill down but i'm open to doing this but I want to have a conversation before adding it ✨ @amritk @geoffgscott

Reply:
So this is the way I think of it

- if data is needed 1-2 children deep, prop drill
- if data is needed deeper than that, but still a descendant, [use provide/inject](https://vuejs.org/guide/components/provide-inject)
- if data is needed in a sibling, or no direct parent/descendant path use a store
- if data need is triggered by an event, use the bus

what you think @geoffgscott, this is more of a style guide thing as they all work

Reply:
Yeah @amritk I agree. I would probably prop drill up to 4 levels but it's preferential. 

While _generally_ I would not agree with this pattern there could be a use case in the future here if we find we are referencing the config object very deeply. I don't think this use case justifies it. we are only drilling from SearchButton through to SearchModal. 

Global stores should really be avoided as much as possible _especially_ when you want to control the store values from external triggers.